spark/src/main/assembly/dependency-reduced.xml

```diff
@@ -34,16 +34,15 @@
       </unpackOptions>
       <scope>runtime</scope>
       <outputDirectory>/</outputDirectory>
-      <useTransitiveFiltering>true</useTransitiveFiltering>
+      <useProjectArtifact>true</useProjectArtifact>
       <includes>
         <!-- guava only included to get Preconditions in mahout-math and mahout-hdfs -->
         <include>com.google.guava:guava</include>
-        <include>com.github.scopt_${scala.compat.version}</include>
+        <include>com.github.scopt:scopt_2.10</include>
+        <include>com.github.scopt:scopt_2.11</include>
         <include>com.tdunning:t-digest</include>
         <include>org.apache.commons:commons-math3</include>
         <include>it.unimi.dsi:fastutil</include>
-        <include>org.apache.mahout:mahout-native-viennacl_${scala.compat.version}</include>
```

Comment:
can do the same thing as is done with scopt here,  ie. include both versions and then use `buildtools/change-scala-versions.sh` to switch?

Reply:
deal- there's no switch happening with change-scala-version.sh though

Reply:
I take it back- viennacl won't be included bc there is no dependency... sorry.

Reply:
I think after we discussed yesterday, if adding these jars to the classpath is the better practice, and we want this jar to be a slim as possible, leaving them out would be fine.   I am unclear on how often its shipped around on the backend.

 I think that the spark method- just adding to the `spark.executor.extraClassPath` and `spark.driver.extraClassPath` makes sense.  @rawkintrevo this is what you were saying we should to, correct and to just add it to the documentation?  I don't think that the `dependency-reduced.jar` is the way we want to be shipping everything in the future.

I think that this is something that we as a project need to discuss at some point.  The original goal for these solver modules (and the somewhat current, hacky implementation) was to probe the classpath for the existence of a solver, and then to use e.g. viennacl-omp#Mmul if it is on the classpath.  Not sure that adding it to the dependency reduce jar is the way to go.  

I say we just ship this as is-  still need to do more testing on a cluster due to pat's next scopt find:

```
Exception in thread "main" java.lang.NoClassDefFoundError: scala/Product
    at java.lang.ClassLoader.defineClass1(Native Method)
    at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
```

Reply:
No we can't ship this with spark-itemsimilarity driver not working. about 50% of the few questions on the list, not to mention the IBM folks are using it.

Reply:
I don't think IBM is using this- they compiled their own code.

Reply:
As I have because we are having trouble here